### ISSUE TITLE
Repeated content

### ISSUE THREAD

**Issue by nguyenvo09** [timbmg/Sentence-VAE#7]
I used your code and trained a model to generate new sentences. The problem is that there are so many repeated tokens in generated samples. 

Any insight how to deal with this? 

For example, token <unk> appears so many times.

> unocal corp . said it reached its agreement with gm and <unk> products and <unk> products <eos>
judge keenan told her conviction of litigation sciences denies that the nsc staff was dismissed and that judge o'kicki had been dismissed <eos>
<unk> <unk> president of the u . s . chamber of commerce department store corp . said it will be able to have to be required to the <unk> of the company ' s <unk> and <unk> <eos>
rep . <unk> d . calif . and <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> and <unk> <unk> to the <unk> of the <unk> <unk> program trading and <unk> <unk> and <unk> <unk> and <unk> and <unk> and <unk> and <unk> and <unk> and
the issue is rated triple-a by moody ' s <unk> <unk> <unk> <unk> and <unk> <unk> <unk> <unk> <unk> of the <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> & co . and <unk> <unk> <unk> <unk> <eos>
in late new york stock exchange composite trading yesterday closed at $ n down n cents <eos>
i ' m ok to get back <eos>
the <unk> of the <unk> <unk> <unk> in the <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> & co . and <unk> <unk> inc . a new york firm based in the <unk> ill . <unk> of <unk> <unk> & co . and <unk> <unk> & co . and
the senate voted to be discussed to the board ' s request for the next few years <eos>
sales rose n n to $ n billion from $ n billion <eos>
<unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> said the <unk> of the <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> and <unk> <unk> that it would be a mistake to prevent the <unk> of the <unk> <eos>
we ' re not going to be <unk> <eos>
and <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> and <unk> <unk> <unk> <unk> and <unk> <unk> <unk> <unk> and <unk> <unk> and <unk> <unk> and <unk> <unk> and <unk> <unk> and <unk> <unk> and <unk> <unk> and <unk>
we ' re shocked the <unk> of the <unk> and <unk> of the <unk> and <unk> of the <unk> and <unk> of the <unk> <eos>
the two sides are n't a n n stake in the first six months <eos>
in the past n years the <unk> is one of the <unk> <unk> <eos>
the <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <eos>
the company said it had n't been reached for comment <eos>
but they are n't yet recommending their own accounts <eos>
the <unk> of the <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> and <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> and <unk> <unk> <unk> <unk> <unk> and <unk> <unk> <unk> <unk> and <unk> <unk> <unk> <unk> and <unk> <unk> <unk> <unk> and <unk> <unk> <unk>
the treasury bills will be sold for the next year <eos>
one of the <unk> of the <unk> <unk> <unk> in the <unk> of the <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> by the <unk> of the <unk> institute of <unk> <unk> a <unk> <unk> who heads the <unk> of the <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk>
but the <unk> of the <unk> <unk> <unk> <unk> <unk> the <unk> of the <unk> <unk> <unk> <unk> <unk> the <unk> of the <unk> <eos>
mr . krenz is <unk> to the <unk> <eos>
in the first nine months of the year norwood had been anticipated by the end of the year ended march n and n n <eos>
and <unk> are <unk> with the exception of the <unk> and <unk> <unk> and <unk> <unk> and <unk> <unk> and <unk> <unk> and <unk> <unk> and <unk> <unk> and <unk> <unk> and <unk> <unk> and <unk> and <unk> and <unk> and <unk> and <unk> and <unk> and <unk> and <unk>
but the <unk> of the <unk> era <eos>

**Comment by timbmg** [comment]
For how long did you train?
What was your final KL/NLL Loss?

**Comment by preke** [comment]
Is that seq2seq-like model you want to implement?
I have the same problem met.
It seems like when training, the input of the decoder also have to be sorted by length.
While in the evaluation part, we do not have prior knowledge of the lengths of the sentences we want to generate, so, this part of the information is kind of lost.
Also, it seems seq2seq-like decoder could only be implemented by RNNLM, is that true?(like the code below:

```python
            t = 0
            while(t < self.max_sequence_length-1):
                if t == 0:
                    input_sequence = Variable(torch.LongTensor([self.sos_idx] * batch_size), volatile=True)
                    if torch.cuda.is_available():
                        input_sequence = input_sequence.cuda()
                        outputs        = outputs.cuda()

                input_sequence  = input_sequence.unsqueeze(1)
                input_embedding = self.embedding(input_sequence) # b * e
                output, hidden  = self.decoder_rnn(input_embedding, hidden) 
                logits          = self.outputs2vocab(output) # b * v
                outputs[:,t,:]  = nn.functional.log_softmax(logits, dim=-1).squeeze(1)  # b * v 
                input_sequence  = self._sample(logits)
                t += 1

            outputs = outputs.view(batch_size, self.max_sequence_length, self.embedding.num_embeddings)
```
